roles/ceph-facts/tasks/devices.yml

```diff
@@ -85,4 +88,5 @@
     - item.value.sectors != "0"
     - item.value.partitions|count == 0
     - item.value.holders|count == 0
+    - ansible_facts['mounts'] | json_query(device_mount_query) | length == 0
```

Comment:
```suggestion
    - item.key not in ansible_facts['mounts'] | map(attribute='device') | list | map('regex_replace', '^.*\\/(.*)$', '\\1') | list
```

Reply:
@guits Nice one. Done

Reply:
@clwluvw
see @dsavineau's approach which is even simpler
```
ansible_facts['mounts'] | selectattr('device', 'match', '^.*'+item.key+'$') | length == 0
```

Reply:
@guits Are you sure about this regex? isn't that better to use `equalto` in `selectattr` and compare it with the `regex_replace` of the disk?
I mean something like this:
```yaml
- name: set_fact devices generate device list when osd_auto_discovery
  vars:
    device: "{{ item.key | regex_replace('^', '/dev/') }}"
  set_fact:
    devices: "{{ devices | default([]) | union([device]) }}"
  with_dict: "{{ ansible_facts['devices'] }}"
  when:
    - osd_auto_discovery | default(False) | bool
    - ansible_facts['devices'] is defined
    - item.value.removable == "0"
    - item.value.sectors != "0"
    - item.value.partitions|count == 0
    - item.value.holders|count == 0
    - ansible_facts['mounts'] | selectattr('device', 'equalto', device) | length == 0
    - item.key is not match osd_auto_discovery_exclude
```

Reply:
that would work too I guess. I'm just trying to remember but if I'm not wrong, we had issues with `equalto` in the past, can't recall very well what it was to be honest... Perhaps @dsavineau remembers

Reply:
@guits This regex I think will match something like `/abcdefg/asd/qwe/zxc/dev/sda` for `/dev/sda` and the `equalto` will resolve this.

Reply:
> > @guits This regex I think will match something like `/abcdefg/asd/qwe/zxc/dev/sda` for `/dev/sda` and the `equalto` will resolve this.
> 
> Do you have an example where you could end up with `/abcdefg/asd/qwe/zxc/dev/sda` in `ansible_facts['mounts'] ` ?

Not now but I thought maybe the right solution could be `equalto` to be sure we match what we expect in general =)

Reply:
I think the reason for not using `equalto` was because of the jinja2 version for python 2.7 on EL 7.
Given RHCS 5 (downstream) isn't supported on EL7 and py27 is eol anyway, I think it's fine using `equalto` now.